sanic/router.py

```diff
@@ -65,17 +65,19 @@ def my_route_with_type(request, my_param: my_type):
     routes_always_check = None
     parameter_pattern = re.compile(r'<(.+?)>')
 
-    def __init__(self):
+    def __init__(self, converters=None):
```

Comment:
I think it's better to make converters mandatory and pass whatever is required in each case.

Reply:
Not all users should redefine that behavior.

Reply:
True, but current implementation breaks DIP design.
And your implementation makes design even harder, since there's some predefined list which you're updating.
Imagine that tomorrow you need another rules or even another order of rules.
*Though this would be a BC break, and should be handled properly during release.*

Reply:
I prefer it as is, since that pattern is common practice in python and throughout Sanic. Though admittedly I had never heard of DIP design before this, so I may be missing the benefit.

Reply:
I can suggest solution with default converter which equal to REGEX_TYPES.